libherokubuildpack/src/env_layer.rs

```diff
@@ -0,0 +1,125 @@
+/// launch (runtime) environments.
+///
+/// Example:
+///
+/// ```no_run
+///# use libcnb::build::{BuildContext, BuildResult, BuildResultBuilder};
+///# use libcnb::data::launch::{LaunchBuilder, ProcessBuilder};
+///# use libcnb::data::process_type;
+///# use libcnb::detect::{DetectContext, DetectResult, DetectResultBuilder};
+///# use libcnb::generic::{GenericError, GenericMetadata, GenericPlatform};
+///# use libcnb::{buildpack_main, Buildpack};
+///# use libcnb::data::layer::LayerName;
+///
+///# pub(crate) struct HelloWorldBuildpack;
+///
+/// use libcnb::Env;
+/// use libcnb::data::layer_name;
+/// use libcnb::layer_env::{LayerEnv, ModificationBehavior, Scope};
+/// use libherokubuildpack::env_layer;
+///
+///# impl Buildpack for HelloWorldBuildpack {
+///#     type Platform = GenericPlatform;
+///#     type Metadata = GenericMetadata;
+///#     type Error = GenericError;
+///
+///#     fn detect(&self, _context: DetectContext<Self>) -> libcnb::Result<DetectResult, Self::Error> {
+///#         todo!()
+///#     }
+///
+///#     fn build(&self, context: BuildContext<Self>) -> libcnb::Result<BuildResult, Self::Error> {
+///         let env = Env::from_current();
+///
+///         let env = {
+///             let layer = context.handle_layer(
+///                 layer_name!("configure_env"),
+///                 env_layer::ConfigureEnvLayer::new(
+///                     LayerEnv::new()
+///                         .chainable_insert(
+///                             Scope::All,
+///                             ModificationBehavior::Override,
+///                             "BUNDLE_GEMFILE", // Tells bundler where to find the `Gemfile`
+///                             context.app_dir.join("Gemfile"),
+///                         )
+///                         .chainable_insert(
+///                             Scope::All,
+///                             ModificationBehavior::Override,
+///                             "BUNDLE_CLEAN", // After successful `bundle install` bundler will automatically run `bundle clean`
+///                             "1",
+///                         )
+///                         .chainable_insert(
+///                             Scope::All,
+///                             ModificationBehavior::Override,
+///                             "BUNDLE_DEPLOYMENT", // Requires the `Gemfile.lock` to be in sync with the current `Gemfile`.
+///                             "1",
+///                         )
+///                         .chainable_insert(
+///                             Scope::All,
+///                             ModificationBehavior::Default,
+///                             "MY_ENV_VAR",
+///                             "Whatever I want",
+///                         ),
+///                 ),
+///             )?;
+///             layer.env.apply(Scope::Build, &env)
+///         };
+///
+///#        todo!()
+///#     }
+///# }
+/// ```
+pub struct ConfigureEnvLayer<B: libcnb::Buildpack> {
```

Comment:
If we end up keeping this (versus a read/write layer concept), we need to align on naming. We already have a `LayerEnv` which this takes, so I didn't want to call it `EnvLayer`. We could use `Set` instead of `Configure`. The module name is meh as well, I'm open to suggestions.